lib/doorkeeper/rails/helpers.rb

```diff
@@ -31,10 +31,16 @@ def doorkeeper_render_error
 
       def doorkeeper_render_error_with(error)
         options = doorkeeper_render_options(error) || {}
+        status = if options.delete(:not_found_for_forbidden) &&
+                    error.status == :forbidden
+                   :not_found
+                 else
+                   error.status
+                 end
         if options.blank?
-          head error.status
+          head status
         else
-          options[:status] = error.status
+          options[:status] = status
           options[:layout] = false if options[:layout].nil?
           render options
         end
```

Comment:
Refactored, with the `delete` staying in `doorkeeper_render_error_with`, so `doorkeeper_status_for_error` doesn't have hidden side effects. `doorkeeper_status_for_error` felt more consistent/descriptive, but if you prefer `http_status_code`, I'm happy to rename it.